c2cgeoportal/scaffolds/update/CONST_alembic/main/versions/116b9b79fc4d_internal_and_external_layer_tables_.py

```diff
@@ -0,0 +1,245 @@
+            ForeignKey(schema + '.layer.id'), primary_key=True
+        ),
+        Column(
+            'server_ogc_id', Integer,
+            ForeignKey(schema + '.server_ogc.id')
+        ),
+        Column('layer', Unicode),
+        Column('style', Unicode),
+        Column('time_mode', Unicode, server_default=u'disabled', nullable=False),
+        Column('time_widget', Unicode, server_default=u'slider', nullable=False),
+        schema=schema,
+    )
+
+    # move data from layer_internal_wms and layer_external_wms to the new
+    # layer_wms and server_ogc tables
+
+    # ocg for internal
+
+    # default 'image/jpeg', 'image/png'
+    op.execute(
+        "INSERT INTO %(schema)s.server_ogc (name, description, type, image_type, auth, wfs_support) "
+        "SELECT 'source for ' || image_type AS name, 'default source for internal ' || image_type AS description, 'mapserver' AS type, image_type, 'main' AS auth, 'true' AS wfs_support "
+        "FROM (SELECT UNNEST(ARRAY['image/jpeg', 'image/png']) AS image_type) AS foo" % {
+            'schema': schema,
+        }
+    )
+    # other custom image types, including NULL
+    op.execute(
+        "INSERT INTO %(schema)s.server_ogc (name, description, type, image_type, auth, wfs_support) "
+        "SELECT 'source for ' || image_type AS name, 'default source for internal ' || image_type "
+        "AS description, 'mapserver' AS type, image_type, 'main' AS auth, 'true' AS wfs_support from ("
+        "SELECT DISTINCT(image_type) FROM %(schema)s.layer_internal_wms "
+        "WHERE image_type NOT IN ('image/jpeg', 'image/png')"
+        ") as foo" % {
+            'schema': schema,
+        }
+    )
+
+    # layers for internal
+
+    # internal with not null image_type
+    op.execute(
+        "INSERT INTO %(schema)s.layer_wms (id, server_ogc_id, layer, style, time_mode, time_widget) "
+        "SELECT lew.id, so.id, layer, style, time_mode, time_widget "
+        "FROM %(schema)s.layer_internal_wms AS lew, %(schema)s.server_ogc AS so WHERE lew.image_type=so.image_type AND so.type IS NOT NULL" % {
+            'schema': schema,
+        }
+    )
+    # internal with null image_type
+    op.execute(
+        "INSERT INTO %(schema)s.layer_wms (id, server_ogc_id, layer, style, time_mode, time_widget) "
+        "SELECT lew.id, so.id, layer, style, time_mode, time_widget "
+        "FROM %(schema)s.layer_internal_wms AS lew, %(schema)s.server_ogc AS so where lew.image_type IS NULL AND so.image_type='image/png'" % {
+            'schema': schema,
+        }
+    )
+
+    # ocg for externals
+    # using temporary original_id column
+    op.execute(
+        "INSERT INTO %(schema)s.server_ogc (name, url, image_type, auth, is_single_tile, original_id) "
+        "SELECT 'source for ' || layer, url, image_type, 'none', "
+        "CASE WHEN is_single_tile IS TRUE THEN TRUE ELSE FALSE END, id "
+        "FROM %(schema)s.layer_external_wms" % {
+            'schema': schema,
+        }
+    )
```

Comment:
We shouldn't have many different entries for same: url, image_type and is_single_tile ...